indigo/indigo/src/main/scala/indigo/platform/renderer/webgl2/RendererWebGL2.scala

```diff
@@ -222,6 +240,72 @@ final class RendererWebGL2(
 
   private given CanEqual[Option[Int], Option[Int]] = CanEqual.derived
 
+  def captureScreen(captureOptions: Batch[ScreenCaptureConfig]): Batch[Either[String, AssetType.Image]] =
+    _prevSceneData match {
+      case null => captureOptions.map(_ => Left("No scene data to capture"))
+      case _ =>
+        val prevSceneData   = _prevSceneData
+        val prevGameRuntime = _prevGameRuntime
+
+        captureOptions.map(option =>
+          val canvas = dom.document.createElement("canvas").asInstanceOf[html.Canvas]
+          val ctx2d =
+            canvas.getContext("2d", cNc.context.getContextAttributes()).asInstanceOf[dom.CanvasRenderingContext2D]
+          val magnifiedClip = option.croppingRect match {
+            case Some(rect) => rect * cNc.magnification
+            case None       => Rectangle(0, 0, screenWidth, screenHeight)
+          }
+          val imageSize = Size(
+            (magnifiedClip.width * option.scale.getOrElse(Vector2.one).x).toInt,
+            (magnifiedClip.height * option.scale.getOrElse(Vector2.one).y).toInt
+          )
+
+          canvas.width = imageSize.width
+          canvas.height = imageSize.height
+          ctx2d.imageSmoothingEnabled = false
+
+          drawScene(
+            ProcessedSceneData(
+              _prevSceneData.layers.filter(l =>
+                l.bindingKey match {
+                  case Some(bk) => option.excludeLayers.exists(_ == bk) == false
+                  case None     => true
+                }
+              ),
+              _prevSceneData.cloneBlankDisplayObjects,
+              _prevSceneData.shaderId,
+              _prevSceneData.shaderUniformData,
+              _prevSceneData.camera
+            ),
+            _prevGameRuntime
+          )
+
+          _prevSceneData = prevSceneData
+          _prevGameRuntime = prevGameRuntime
+
+          ctx2d.drawImage(
+            cNc.canvas,
+            magnifiedClip.x,
+            magnifiedClip.y,
+            magnifiedClip.width,
+            magnifiedClip.height,
+            0,
+            0,
+            imageSize.width,
+            imageSize.height
+          )
+          val dataUrl = canvas.toDataURL(option.imageType.toString())
+          canvas.remove()
+
+          Right(
+            AssetType.Image(
+              AssetName(option.name.getOrElse(s"capture-${System.currentTimeMillis()}")),
+              AssetPath(dataUrl)
+            )
+          )
+        )
+    }
```

Comment:
This looks good. So if I'm reading correctly, the process is:

1. Take the previous scene (so that you can filter out the layers people don't want to capture).
2. Set up a new canvas.
3. Re-render which effectively overwrites the usual canvas context.
4. Grab it, and stuff it into the new canvas.
5. Save it.

Have I got that right?